tensorzero-core/src/endpoints/batch_inference.rs

```diff
@@ -244,16 +236,21 @@ pub async fn start_batch_inference_handler(
     // TODO (#496): remove this extra clone
     // Spent a while fighting the borrow checker here, gave up
     // The issue is that inference_config holds the ToolConfigs and ModelInferenceRequest has lifetimes that conflict with the inference_config
-    let cloned_config = inference_config.clone();
-    let inference_configs = cloned_config.inference_configs(&episode_ids, &inference_ids);
-    while !candidate_variant_names.is_empty() {
+    while !candidate_variants.is_empty() {
         // We sample the same variant for the whole batch
         let (variant_name, variant) = sample_variant(
-            &mut candidate_variant_names,
-            function.variants(),
+            &mut candidate_variants,
             &params.function_name,
             first_episode_id,
         )?;
+        let inference_config = BatchInferenceConfig::new(
```

Comment:
this does involve looping over the batch to generate tool configs inside so there's probably a hit to performance from this if we actually do variant sampling for batches. I don't think this is something people do on purpose though.